internal/hcl/modules/s3_cache.go

```diff
@@ -55,19 +56,32 @@ func (cache *S3Cache) applyPrefix(key string) string {
 // Exists checks if the key exists in the S3 bucket
 func (cache *S3Cache) Exists(key string) (bool, error) {
 	prefixedKey := cache.applyPrefix(key)
-	_, err := cache.s3Client.HeadObject(&s3.HeadObjectInput{
+	headObj, err := cache.s3Client.HeadObject(&s3.HeadObjectInput{
 		Bucket: aws.String(cache.bucketName),
 		Key:    aws.String(prefixedKey),
 	})
 	if err != nil {
 		if awsErr, ok := err.(awserr.Error); ok {
 			switch awsErr.Code() {
-			case s3.ErrCodeNoSuchKey, "NotFound":
+			case s3.ErrCodeNoSuchKey, "NotFound", "Forbidden":
```

Comment:
Should we add a case for forbidden and log it out. Seems more of a failure case than just it not being in the cache?

Reply:
I thought about that, but I think this will just spam the logs. If you don't have `ListBucket` permission which is what we do then every case of NotFound shows as Forbidden so there's no way of differentiating.